src/block.ts

```diff
@@ -45,8 +65,11 @@ class Block {
     return block;
   }
 
+  getHash(): Buffer;
+  getHash<T extends boolean>(hexStr: T): T extends true ? string : Buffer;
   getHash(hexStr = false) {
     if (!this.hash) {
+      if (!this.header) throw Error("Missing header");
```

Comment:
@kevinejohn I am throwing a lot of errors throughout this class because `this.header` could undefined. Do you see a problem with that?